cpp/src/lance/arrow/scanner.cc

```diff
@@ -33,6 +33,8 @@ void ScannerBuilder::Project(const std::vector<std::string>& columns) { columns_
 
 void ScannerBuilder::Filter(const ::arrow::compute::Expression& filter) { filter_ = filter; }
 
+void ScannerBuilder::BatchSize(int64_t batch_size) { batch_size_ = batch_size; }
```

Comment:
What are valid values for batch_size? Do we need to validate that here or is it better to do it somewhere else?

Reply:
Good point. Lemme just change the interface to resemble `arrow::dataset::ScannBuilder` then.

https://arrow.apache.org/docs/cpp/api/dataset.html#scanning